code/game/machinery/pipe/construction.dm

```diff
@@ -147,14 +147,16 @@ Buildable meters
 /obj/item/pipe/attack_self(mob/user)
 	return rotate()
 
+GLOBAL_LIST_INIT(pipe_construction_cache, list())
+
 /obj/item/pipe/proc/get_pipe_cache(type, direction)
-	var/static/list/obj/machinery/atmospherics/check_cache
+	var/list/obj/machinery/atmospherics/check_cache = GLOB.pipe_construction_cache
 	if(!islist(check_cache))
 		check_cache = list()
 	if(!check_cache[type])
 		check_cache[type] = list()
 	if(!check_cache[type]["[direction]"])
-		check_cache[type]["[direction]"] = new type(null, null, direction)
+		check_cache[type]["[direction]"] = new type(null, FALSE, direction)

```

Comment:
while you're at it you should give them an identifiable name, something along the lines of
```
var/obj/machinery/atmospherics/A = new type(null, FALSE, direction)
A.name = "(CACHE) [A.name]"
check_cache[type]["[direction]"] = A
```

Reply:
kk ok it!